### PR TITLE
update branch naming

### PR DESCRIPTION
**Pull request by jd-lara**
Before fixing the broken tests, I would like to know if this PR is acceptable. The Branch names are currently based on the order that we parse the power models dict. As a result, the names change between runs sometimes and it makes hard to make code based on branch naming.

### CODE REVIEW

File: src/parsers/power_models_data.jl [hunk 1]
```diff
@@ -583,13 +592,13 @@ function read_branch!(sys::System, data, bus_number_to_bus::Dict{Int, Bus}; kwar
         return
     end
 
-    _get_name = get(kwargs, :branch_name_formatter, _get_pm_dict_name)
+    _get_name = get(kwargs, :branch_name_formatter, _get_pm_branch_name)
```
Comment (daniel-thom) [inline src/parsers/power_models_data.jl:595]: 1. It seems problematic if we are inferring information from the name. We should just need them to be unique. There's probably a detail I'm missing.
2. This same function is used in many other places. Notably, `make_dcline`. Is this a problem anywhere else?

Comment (jd-lara) [reply src/parsers/power_models_data.jl:595]: Re 1. It is common to use names to refer to lines. In a large system, operators use names and numbers, for instance, see the print screen with real line names. 

image

Re 2. true, DC Line needs the same change.

**Review by claytonpbarrows — APPROVED** [review]
Generally, I think this is an improvement. Previously, we were inferring an identifier on branch circuits based on the order that we were encountering them, right? I vaguely recall adding the iteration index to them to ensure uniqueness, but this is better.

File: src/parsers/power_models_data.jl [hunk 2]
```diff
@@ -71,6 +71,14 @@ function _get_pm_dict_name(device_dict)
     return get(device_dict, "name", string(device_dict["index"]))
 end
 
+"""
+Internal branch name retreval from pm2ps_dict
+"""
+function _get_pm_branch_name(device_dict, bus_f, bus_t)
+    index = get(device_dict, "name", device_dict["index"])
+    return string("$(get_name(bus_f))-$(get_name(bus_t))-i_$index")
```
Comment (daniel-thom) [inline src/parsers/power_models_data.jl:79]: You can remove the call to `string`

Comment (jd-lara) [reply src/parsers/power_models_data.jl:79]: thanks

**Review by daniel-thom — APPROVED** [review]

**Pull request merged**